File: app/src/main/java/com/moneycollect/example_java/activity/PaymentExampleActivity.java
package com.moneycollect.example_java.activity;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.Nullable;

import com.google.gson.Gson;
import com.moneycollect.android.model.request.RequestConfirmPayment;
import com.moneycollect.android.model.request.RequestCreatePayment;
import com.moneycollect.android.model.response.Payment;
import com.moneycollect.android.net.net.ApiResultCallback;
import com.moneycollect.android.utils.MoneyCollectButtonUtils;
import com.moneycollect.example.R;
import com.moneycollect.example.databinding.ActivityCreatePaymentLayoutBinding;
import com.moneycollect.example_java.BaseExampleActivity;
import com.moneycollect.example_java.TestRequestData;
import com.moneycollect.example_java.utils.TempUtils;

import org.jetbrains.annotations.NotNull;

import java.math.BigInteger;

/**
 * [PaymentExampleActivity] show create Payment、retrieve Payment and confirm Payment  sample
 */
public class PaymentExampleActivity extends BaseExampleActivity implements View.OnClickListener{

    private ActivityCreatePaymentLayoutBinding viewBinding;

    private Button createPaymentBtn;

    private Button retrievePaymentBtn;

    private Button confirmPaymentBtn;

    private TextView resultTv;
    private TextView resultTag;

    //payment for pay
    private Payment payment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        viewBinding = ActivityCreatePaymentLayoutBinding.inflate(getLayoutInflater());
        setContentView(viewBinding.getRoot());
        initUi();
    }

    private void initUi() {
        resultTv=viewBinding.resultTv;
        resultTag=viewBinding.resultTag;

        createPaymentBtn=viewBinding.mcCreatePaymentBtn;
        createPaymentBtn.setOnClickListener(this);

        retrievePaymentBtn=viewBinding.mcRetrievePaymentBtn;
        retrievePaymentBtn.setOnClickListener(this);

        confirmPaymentBtn=viewBinding.mcConfirmPaymentBtn;
        confirmPaymentBtn.setOnClickListener(this);
    }

    /**
     * create Payment
     */
    private void createPayment() {
        RequestCreatePayment createPayment = TestRequestData.Companion.getTestRequestPayment();
        if (createPayment.getConfirmationMethod() == RequestCreatePayment.ConfirmationMethod.Automatic){
            if (TextUtils.isEmpty(createPayment.getPaymentMethod())){
                showToast(getString(R.string.payment_method_empty_str));
                return;
            }
        }
        showLoadingDialog();
        moneyCollect.createPayment(createPayment, new ApiResultCallback<Payment>() {
                    @Override
                    public void onSuccess(@NotNull Payment result) {
                        dismissLoadingDialog();
                        payment=result;
                        resultTag.setText(R.string.create_payment_success_str);
                        resultTv.setText(TempUtils.formatString(new Gson().toJson(result)));
                    }

                    @Override
                    public void onError(@NotNull Exception e) {
                        dismissLoadingDialog();
                        resultTag.setText(R.string.create_payment_error_str);
                        resultTv.setText(TempUtils.formatString(new Gson().toJson(e)));
                    }
                }
        );
    }

    /**
     * confirm Payment
     */
    private void confirmPayment() {
        if (TextUtils.isEmpty(payment.getPaymentMethod())) {
            showToast(getString(R.string.payment_method_empty_str));
            return;
        }
        if (TextUtils.isEmpty(payment.getId())) {
            showToast(getString(R.string.payment_id_empty_str));
            return;
        }
        if (TextUtils.isEmpty(payment.getClientSecret())) {
            showToast(getString(R.string.payment_client_secret_empty_str));
            return;
        }
        showLoadingDialog();
        RequestConfirmPayment requestConfirmPayment = new RequestConfirmPayment(
                BigInteger.valueOf(payment.getAmount()),
                payment.getCurrency(),
                payment.getId(),
                payment.getIp(),
                payment.getNotifyUrl(),
                payment.getPaymentMethod(),
                payment.getReceiptEmail(),
                payment.getReturnUrl(),
                payment.getSetupFutureUsage(),
                new RequestConfirmPayment.Shipping(
                        TestRequestData.Companion.getAddress(),
                        TestRequestData.Companion.getFirstName(),
                        TestRequestData.Companion.getLastName(),
                        TestRequestData.Companion.getPhone()
                ),
               TestRequestData.Companion.getWebsite()
        );
        moneyCollect.confirmPayment(requestConfirmPayment, payment.getClientSecret(),
                new ApiResultCallback<Payment>() {
                    @Override
                    public void onSuccess(@NotNull Payment result) {
                        dismissLoadingDialog();
                        resultTag.setText(R.string.confirm_payment_success_str);
                        resultTv.setText(TempUtils.formatString(new Gson().toJson(result)));
                    }

                    @Override
                    public void onError(@NotNull Exception e) {
                        dismissLoadingDialog();
                        resultTag.setText(R.string.confirm_payment_error_str);
                        resultTv.setText(TempUtils.formatString(new Gson().toJson(e)));
                    }
                }
        );
    }

    /**
     * retrieve Payment
     */
    private void retrievePayment() {
        if (TextUtils.isEmpty(payment.getId())) {
            showToast(getString(R.string.payment_id_empty_str));
            return;
        }

        showLoadingDialog();
        moneyCollect.retrievePayment(payment.getId(), payment.getClientSecret(),
                new ApiResultCallback<Payment>() {
                    @Override
                    public void onSuccess(@NotNull Payment result) {
                        dismissLoadingDialog();
                        resultTag.setText(R.string.retrive_payment_success_str);
                        resultTv.setText(TempUtils.formatString(new Gson().toJson(result)));
                    }

                    @Override
                    public void onError(@NotNull Exception e) {
                        dismissLoadingDialog();
                        resultTag.setText(R.string.retrieve_payment_error_str);
                        resultTv.setText(TempUtils.formatString(new Gson().toJson(e)));
                    }
                }
        );
    }

    /**
     * click event
     */
    @Override
    public void onClick(View view) {
        if (view!=null) {
            if (!MoneyCollectButtonUtils.INSTANCE.isFastDoubleClick(view.getId(), 800)) {
                if (view.getId() ==  R.id.mc_create_payment_btn){
                    createPayment();
                }else if (view.getId() ==  R.id.mc_confirm_payment_btn){
                    if (payment ==null ){
                        showToast(getString(R.string.payment_empty_str));
                        return;
                    }
                    if (TextUtils.isEmpty(payment.getId())){
                        showToast(getString(R.string.payment_id_empty_str));
                        return;
                    }
                    confirmPayment();
                }else if (view.getId() ==  R.id.mc_retrieve_payment_btn){
                    if (payment ==null ){
                        showToast(getString(R.string.payment_empty_str));
                        return;
                    }
                    if (TextUtils.isEmpty(payment.getId())){
                        showToast(getString(R.string.payment_id_empty_str));
                        return;
                    }
                    retrievePayment();
                }
            }
        }
    }
}

File: app/src/main/java/com/moneycollect/example/activity/PaymentExampleActivity.kt
package com.moneycollect.example.activity

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.Button
import android.widget.TextView
import com.google.gson.Gson
import com.moneycollect.android.model.request.RequestConfirmPayment
import com.moneycollect.android.model.request.RequestCreatePayment
import com.moneycollect.android.model.response.Payment
import com.moneycollect.android.net.net.ApiResultCallback
import com.moneycollect.android.utils.MoneyCollectButtonUtils
import com.moneycollect.example.BaseExampleActivity
import com.moneycollect.example.R
import com.moneycollect.example.TestRequestData
import com.moneycollect.example.databinding.ActivityCreatePaymentLayoutBinding
import com.moneycollect.example.utils.formatString

/**
 * [PaymentExampleActivity] show create Payment、retrieve Payment and confirm Payment  sample
 */
class PaymentExampleActivity : BaseExampleActivity(), View.OnClickListener {

    private var viewBinding: ActivityCreatePaymentLayoutBinding? = null

    private var createPaymentBtn: Button? = null

    private var retrievePaymentBtn: Button? = null

    private var confirmPaymentBtn: Button? = null

    private var resultTv: TextView? = null
    private var resultTag: TextView? = null

    //payment for pay
    private var payment: Payment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewBinding = ActivityCreatePaymentLayoutBinding.inflate(layoutInflater)
        setContentView(viewBinding!!.root)
        initUi()
    }

    private fun initUi() {
        resultTv = viewBinding?.resultTv
        resultTag = viewBinding?.resultTag

        createPaymentBtn = viewBinding?.mcCreatePaymentBtn
        createPaymentBtn?.setOnClickListener(this)

        retrievePaymentBtn = viewBinding?.mcRetrievePaymentBtn
        retrievePaymentBtn?.setOnClickListener(this)

        confirmPaymentBtn = viewBinding?.mcConfirmPaymentBtn
        confirmPaymentBtn?.setOnClickListener(this)
    }

    /**
     * create Payment
     */
    private fun createPayment() {
        val createPayment = TestRequestData.testRequestPayment
        if (createPayment.confirmationMethod == RequestCreatePayment.ConfirmationMethod.Automatic) {
            if (TextUtils.isEmpty(createPayment.paymentMethod)) {
                showToast(getString(R.string.payment_method_empty_str))
                return
            }
        }
        showLoadingDialog()
        moneyCollect.createPayment(createPayment,
            object : ApiResultCallback<Payment> {
                override fun onSuccess(result: Payment) {
                    dismissLoadingDialog()
                    payment = result
                    resultTag?.text = getString(R.string.create_payment_success_str)
                    resultTv?.text = formatString(Gson().toJson(result))
                }

                override fun onError(e: Exception) {
                    dismissLoadingDialog()
                    resultTag?.text =getString(R.string.create_payment_error_str)
                    resultTv?.text = formatString(Gson().toJson(e))
                }
            })
    }

    /**
     * confirm Payment
     */
    private fun confirmPayment() {
        if (TextUtils.isEmpty(payment?.paymentMethod)) {
            showToast(getString(R.string.payment_method_empty_str))
            return
        }
        if (TextUtils.isEmpty(payment?.id)) {
            showToast(getString(R.string.payment_id_empty_str))
            return
        }
        if (TextUtils.isEmpty(payment?.clientSecret)) {
            showToast(getString(R.string.payment_client_secret_empty_str))
            return
        }
        showLoadingDialog()
        val requestConfirmPayment = RequestConfirmPayment(
            amount = payment?.amount?.toBigInteger(),
            currency = payment?.currency,
            id = payment?.id,
            ip = payment?.ip,
            notifyUrl = payment?.notifyUrl,
            paymentMethod = payment?.paymentMethod,
            receiptEmail = payment?.receiptEmail,
            returnUrl = payment?.returnUrl,
            setupFutureUsage = payment?.setupFutureUsage,
            shipping = RequestConfirmPayment.Shipping(
                TestRequestData.address,
                firstName = TestRequestData.firstName,
                lastName = TestRequestData.lastName,
                phone = TestRequestData.phone
            ),
            website = TestRequestData.website
        )
        moneyCollect.confirmPayment(requestConfirmPayment, payment?.clientSecret,
            object : ApiResultCallback<Payment> {
                override fun onSuccess(result: Payment) {
                    dismissLoadingDialog()
                    resultTag?.text = getString(R.string.confirm_payment_success_str)
                    resultTv?.text = formatString(Gson().toJson(result))
                }

                override fun onError(e: Exception) {
                    dismissLoadingDialog()
                    resultTag?.text = getString(R.string.confirm_payment_error_str)
                    resultTv?.text = formatString(Gson().toJson(e))
                }
            })
    }

    /**
     * retrieve Payment
     */
    private fun retrievePayment() {
        if (TextUtils.isEmpty(payment?.id)) {
            showToast(getString(R.string.payment_id_empty_str))
            return
        }
        payment?.id?.let {
            showLoadingDialog()
            moneyCollect.retrievePayment(it, payment?.clientSecret,
                object : ApiResultCallback<Payment> {
                    override fun onSuccess(result: Payment) {
                        dismissLoadingDialog()
                        resultTag?.text = getString(R.string.retrive_payment_success_str)
                        resultTv?.text = formatString(Gson().toJson(result))
                    }

                    override fun onError(e: Exception) {
                        dismissLoadingDialog()
                        resultTag?.text = getString(R.string.retrieve_payment_error_str)
                        resultTv?.text = formatString(Gson().toJson(e))
                    }
                })
        }
    }

    /**
     * click event
     */
    override fun onClick(view: View?) {
        if (view != null) {
            if (!MoneyCollectButtonUtils.isFastDoubleClick(view.id, 800)) {
                when (view.id) {
                    R.id.mc_create_payment_btn -> {
                        createPayment()
                    }
                    R.id.mc_confirm_payment_btn -> {
                        if (payment == null) {
                            showToast(getString(R.string.payment_empty_str))
                            return
                        }
                        if (TextUtils.isEmpty(payment?.id)) {
                            showToast(getString(R.string.payment_id_empty_str))
                            return
                        }
                        confirmPayment()
                    }
                    R.id.mc_retrieve_payment_btn -> {
                        if (payment == null) {
                            showToast(getString(R.string.payment_empty_str))
                            return
                        }
                        if (TextUtils.isEmpty(payment?.id)) {
                            showToast(getString(R.string.payment_id_empty_str))
                            return
                        }
                        retrievePayment()
                    }
                }
            }
        }
    }
}